pkgs/development/compilers/pakcs/default.nix

```diff
@@ -1,139 +1,72 @@
-        mv ${name} ${name}.orig
-        ln -s ${name}.orig/frontend/curry-base ${name}
-      '';
-      doCheck = false;
-    }
-  ) {};
-
-  curryFront = haskellPackages.callPackage (
-    { mkDerivation, base, Cabal, containers, directory
-    , filepath, mtl, network-uri, process, syb, transformers
-    }:
-    mkDerivation {
-      pname = "curry-frontend";
-      version = "0.4.1";
-      src = fsrc;
-      isLibrary = true;
-      isExecutable = true;
-      libraryHaskellDepends = [
-        base containers curryBase directory filepath mtl network-uri
-        process syb transformers
-      ];
-      executableHaskellDepends = [
-        base containers curryBase directory filepath mtl network-uri
-        process syb transformers
-      ];
-      testHaskellDepends = [ base Cabal curryBase filepath ];
-      homepage = http://curry-language.org;
-      description = "Compile the functional logic language Curry to several intermediate formats";
-      license = "unknown";
-
-      postUnpack = ''
-        mv ${name} ${name}.orig
-        ln -s ${name}.orig/frontend/curry-frontend ${name}
-      '';
-      doCheck = false;
-    }
-  ) {};
-
-  src = fsrc;
-
-  buildInputs = [ swiPrologLocked makeWrapper glibcLocales rlwrap tk which ];
-
-  patches = [
-    ./adjust-buildsystem.patch
-    ./case-insensitive.patch
+  }).curry-frontend;
+in stdenv.mkDerivation {
+  inherit name src;
+
+  buildInputs = [ swiProlog ];
+  nativeBuildInputs = [ which makeWrapper ];
+
+  makeFlags = [
+    "CURRYFRONTEND=${curry-frontend}/bin/curry-frontend"
+    "DISTPKGINSTALL=yes"
+    # Not needed, just to make script pass
+    "CURRYTOOLSDIR=0"
+    "CURRYLIBSDIR=0"
   ];
 
-  configurePhase = ''
-    # Phony HOME.
-    mkdir phony-home
-    export HOME=$(pwd)/phony-home
+  preConfigure = ''
+    # Since we can't expand $out in `makeFlags`
+    #makeFlags="$makeFlags PAKCSINSTALLDIR=$out/pakcs"
 
-    # SWI Prolog
-    sed -i 's@SWIPROLOG=@SWIPROLOG='${swiPrologLocked}/bin/swipl'@' scripts/pakcsinitrc.sh
+    substituteInPlace currytools/cpm/src/CPM/Repository.curry \
+      --replace "/bin/rm" "rm"
   '';
 
+  # cypm new: EXISTENCE ERROR: source_sink
+  # "/tmp/nix-build-pakcs-2.0.2.drv-0/pakcs-2.0.2/currytools/cpm/templates/LICENSE"
+  # does not exist
   buildPhase = ''
-    # Some comments in files are in UTF-8, so include the locale needed by GHC runtime.
-    export LC_ALL=en_US.UTF-8
-
-    # PAKCS must be build in place due to embedded filesystem references placed by swi.
-
-    # Prepare PAKCSHOME directory.
-    mkdir -p $out/pakcs/bin
-
-    # Set up link to cymake, which has been built already.
-    ln -s ${curryFront}/bin/cymake $out/pakcs/bin/
-    rm -r frontend
-
-    # Prevent embedding the derivation build directory as temp.
-    export TEMP=/tmp
-
-    # Copy to in place build location and run the build.
+    mkdir -p $out/pakcs
     cp -r * $out/pakcs
-    (cd $out/pakcs ; make)
+    (cd $out/pakcs ; make -j$NIX_BUILD_CORES $makeFlags)
```

Comment:
What's this needed for exactly? Does a normal `make` not work?

Reply:
@Infinisil  It works, but swiProlog still embedds references to build directory, so some utilities doesn't work. See upper comment:
```
cypm new: EXISTENCE ERROR: source_sink "/tmp/nix-build-pakcs-2.0.2.drv-0/pakcs-2.0.2/currytools/cpm/templates/LICENSE" does not exist
```